src/main/webapp/dev/js/common/questionMcq.js

```diff
@@ -37,54 +55,166 @@ function addMcqOption(questionNum) {
+ * then sets the required attribute for other weight cell too.
+ */
+function setRequiredAttributeForMcqWeightCells($weightColumn, questionNum, isRequired) {
+    const $weightCells = $weightColumn.find('input[id^="mcqWeight"]');
+    const $otherWeightCell = $(`#mcqOtherWeight-${questionNum}`);
+    const isOtherOptionEnabled = $(`#mcqOtherOptionFlag-${questionNum}`).prop('checked');
+
+    $weightCells.each(function () {
+        $(this).prop('required', isRequired);
+    });
+
+    // If 'other' option is checked, make other weight required.
+    if (isOtherOptionEnabled) {
+        $otherWeightCell.prop('required', isRequired);
+    }
+}
+
+/**
+ * Hides the weight cells and weight label on top of the cells when the checkbox is unchecked,
+ * otherwise, shows the weight cells and label.
+ * If 'other' option is enabled, and 'choices are weighted' checkbox is checked, shows the 'other' weight cell,
+ * otherwise if the checkbox is unchecked, hides the 'other' weight cell.
+ * @param $checkbox
+ * @param questionNum
+ */
+function toggleMcqAssignWeights($checkbox, questionNum) {
+    // The weight label
+    const $weightLabel = $checkbox.parent().siblings('div');
+    const $weightColumn = $(`#mcqWeights-${questionNum}`);
+    const $otherWeightCell = $(`#mcqOtherWeight-${questionNum}`);
+    const $otherEnabledCheckbox = $(`#mcqOtherOptionFlag-${questionNum}`);
+
+    if ($checkbox.prop('checked')) {
+        $weightLabel.show();
+        $weightColumn.show();
+
+        if ($otherEnabledCheckbox.prop('checked')) {
+            $otherWeightCell.show();
+        }
+        // Set the weight cells as required.
+        setRequiredAttributeForMcqWeightCells($weightColumn, questionNum, true);
+    } else {
+        $weightLabel.hide();
+        // Set the weight cells as not required.
+        setRequiredAttributeForMcqWeightCells($weightColumn, questionNum, false);
+        $weightColumn.hide();
+        // If weight is not assigned, no need to check if the other option is checked,
+        // As, other weight will be disabled anyway.
+        $otherWeightCell.hide();
+    }
+}
+
 function changeMcqGenerateFor(questionNum) {
     $(`#mcqGeneratedOptions-${questionNum}`).val($(`#mcqGenerateForSelect-${questionNum}`).prop('value'));
 }
 
 function toggleMcqGeneratedOptions(checkbox, questionNum) {
     if ($(checkbox).prop('checked')) {
-        $(`#mcqChoiceTable-${questionNum}`).find('input[type=text]').prop('disabled', true);
+        $(`#mcqChoices-${questionNum}`).find('input[type=text]').prop('disabled', true);
+        $(`#mcqWeights-${questionNum}`).find('input[type=number]').prop('disabled', true);
+        $(`#mcqOtherWeight-${questionNum}`).prop('disabled', true);
         $(`#mcqChoiceTable-${questionNum}`).hide();
+        $(`#mcqAssignWeights-${questionNum}`).parent().hide();
+        // Hide the 'Weights' label
+        $(`#mcqAssignWeights-${questionNum}`).parent().siblings('div').hide();
```

Comment:
@xpdavid That is because, when we change from generate option to regular mcq choices, all the values like weights, choices get reset to default, and if I bring the label back in the else, then, the label will show when it shouldn't.

 And  this case is covered anyway, as in the else block, I am calling `toggleMcqAssignWeights($(`#mcqAssignWeights-${questionNum}`), questionNum);` which will automatically show the weight label if the checkbox is checked.